src/service/search/cluster/flight.rs

```diff
@@ -901,7 +890,24 @@ pub async fn get_inverted_index_file_list(
         index_stream_name, search_condition,
     );
 
+    // for contains, we need to search all prefixes, for others we can only
+    // search specific prefix
+    let prefix_list = match cfg.common.full_text_search_type.as_str() {
+        "contains" => vec![],
+        _ => terms
+            .iter()
+            .map(|t| {
+                t.to_ascii_lowercase()
+                    .chars()
+                    .next()
+                    .unwrap_or('_')
+                    .to_string()
+            })
+            .collect(),
+    };
```

Comment:
I made this logic based on the logic above :
```rust
let fts_condition = terms
        .iter()
        .map(|x| match cfg.common.full_text_search_type.as_str() {
            "eq" => format!("term = '{x}'"),
            "contains" => format!("term LIKE '%{x}%'"),
            _ => format!("term LIKE '{x}%'"),
        })
        .collect::<Vec<_>>()
        .join(" OR ");
```

My reasoning is that for `contains`, we need to check every term, as the pattern we are looking for can be anywhere, so we don't specify any prefix. In any other cases, the term we are looking for will start with a specific character, so we can pass that as the prefix.

Please check if this is correct reasoning or not.